src/actions/clusterActions.js

```diff
@@ -348,7 +349,49 @@ export function clusterInstallApp(app, clusterID) {
       }
     });
 
+    const optionalCreateAppSecret = new Promise((resolve, reject) => {
+      if (Object.keys(app.secretsYAML).length !== 0) {
+        // If we have user config that we want to create, then
+        // fire off the call to create it.
+        appSecretsApi
+          .createClusterAppSecret(clusterID, app.name, {
+            body: app.secretsYAML,
+          })
+          .then(() => {
+            // The call succeeded, resolve the promise
+            resolve();
+          })
+          .catch(error => {
+            if (error.status === 409) {
+              new FlashMessage(
+                `The Secret for <code>${app.name}</code> already exists on cluster <code>${clusterID}</code>`,
+                messageType.ERROR,
+                messageTTL.LONG
+              );
+            } else if (error.status === 400) {
+              new FlashMessage(
+                `Your Secret appears to be invalid. Please make sure all fields are filled in correctly.`,
+                messageType.ERROR,
+                messageTTL.LONG
+              );
+            } else {
+              new FlashMessage(
+                `Something went wrong while trying to create the Secret. Please try again later or contact support: support@giantswarm.io`,
+                messageType.ERROR,
+                messageTTL.LONG
+              );
+            }
+
+            reject(error);
+          });
+      } else {
+        // Otherwise, no user config, so don't do anything.
+        resolve({});
+      }
+    });
```

Comment:
This is too much for me! :-)
I am ok with not refactoring some of the things we have, but we could use `async/await` for new code and as a way to avoid too much nesting.
It's hard when something fails and you have to debug things like this.

If you need to return a promise, as it seems to be the case, I would write this probably this way:

```javascript
const optionalCreateAppSecret = Object.keys(app.valuesYAML).length === 0
    ? Promise.resolve()
    : await appSecretsApi
          .createClusterAppSecret(clusterID, app.name, {
               body: app.secretsYAML,
           })
          .catch(functionThatHandlesTheseErrrors(error);
```

Reply:
`functionThatHandlesTheseErrrors` makes sense here since I think you are using the same errors twice. 

This code above is something that I find easy to read (I know, this is subjective...)

Reply:
Thx for the help here! 

It's been broken down further now.